pkg/api/apikey.go

```diff
@@ -68,7 +68,10 @@ func (hs *HTTPServer) AddAPIKey(c *models.ReqContext, cmd models.AddApiKeyComman
 		if err == models.ErrInvalidApiKeyExpiration {
 			return Error(400, err.Error(), nil)
 		}
-		return Error(500, "Failed to add API key", err)
+		if err.Error() == models.ErrDuplicateApiKey.Error() {
```

Comment:
Following my suggestion for using `dialect.IsUniqueConstraintViolation` and returning the actual error this can be changed to:
`if err == models.ErrDuplicateApiKey {`